File: java/src/test/stores/WaterStoreTest.java
package test.stores;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;
import static org.junit.Assert.Equals;
import static org.junit.Assert.assertTrue;
import static org.junit.Before;
import static org.junit.Test;

/**
 * Water Store Test class.
 */
public class WaterStoreTest {
    private WaterStore waterStore;
    private Cone cone;

    /**
     * Set up method.
     */
    @Before
    public void SetUp(){
        waterStore = new WaterStore();
        cone = new Cone();
    }
    /**
     * Water bases ice cream test.
     */
    @Test
    public void WaterTest() {
        cone = waterStore.prepareCone("Guava");
        assertEquals("Guava Water Ice Cream", cone.getIceCream().getName());
        cone = waterStore.prepareCone("Mango");
        assertEquals("Mango Water Ice Cream", cone.getIceCream().getName());
        cone = waterStore.prepareCone("Pineapple");
        assertEquals("Pineapple Water Ice Cream", cone.getIceCream().getName());
        cone = waterStore.prepareCone("Strawberry");
        assertEquals("Strawberry Water Ice Cream", cone.getIceCream().getName());
    }
}

File: csharp/factory_method/test/stores/WaterStoreTest.cs
using factory_method.main;
using factory_method.main.stores;
using NUnit.Framework;
using System;
using System.Collections.Generic;
using System.Text;

namespace factory_method.test.stores
{
    class WaterStoreTest
    {
        WaterStore waterStore;
        Cone cone;

        [SetUp]
        public void SetUp()
        {
            waterStore = new WaterStore();
            cone = new Cone();
        }
        [TestCase]
        public void MilkTest()
        {
            cone = waterStore.prepareCone("Guava");
            Assert.AreEqual("Guava Water Ice Cream", cone.getIceCream().getName());
            cone = waterStore.prepareCone("Mango");
            Assert.AreEqual("Mango Water Ice Cream", cone.getIceCream().getName());
            cone = waterStore.prepareCone("Pineapple");
            Assert.AreEqual("Pineapple Water Ice Cream", cone.getIceCream().getName());
            cone = waterStore.prepareCone("Strawberry");
            Assert.AreEqual("Strawberry Water Ice Cream", cone.getIceCream().getName());
        }
    }
}
